src/IO/SensorLogger.cpp

```diff
@@ -42,25 +42,50 @@ void SensorLogger::run()
         boost::filesystem::create_directories(path);
     }
 
-    // bool shouldWriteHeader = !std::experimental::filesystem::exists(path +
-    // filename); fileStream = std::make_shared<std::ofstream>(path + filename,
-    // std::ios_base::app);
-
-    // if (shouldWriteHeader)
-    // {
-    // 	std::ofstream fileStream;
-    // 	fileStream.open(path + filename, std::ios_base::app);
-    // 	writeHeader(fileStream);
-    // 	fileStream.close();
-    // }
+    std::string filename = std::to_string(bootId) + ext;
+    bool shouldWriteHeader = !boost::filesystem::exists(path + filename);
+    std::ofstream fileStream{path + std::to_string(bootId) + ext, std::ios_base::ate};
 
     status.fileStatus = READY;
-    std::ofstream fileStream{path + std::to_string(bootId) + ext, std::ios_base::ate};
+
+    bool isFirstLine = true;
+
     while (true)
     {
         if (!logQueue.empty())
         {
-            dequeueToFile(fileStream);
+
+            // get data
+            sensorsData currentState;
```

Comment:
The get data part should be refactored into a function that returns the currentState

Reply:
sensorsData currentState = getCurrentState();